Perform a complete environment refresh for Studio install, build, deploy

joanise opened the pull request:
<!-- PR template: please provide enough information to guide your reviewers.
Please read Contributing.md before submitting a PR.  -->

### PR Goal? <!-- Explain the main objective of this PR. -->

Big jump in all the environment management for ReadAlongs/Studio:
 - Replace soon-to-be-deprecated `setup.py` and `requirements*.txt` by `pyproject.toml`
 - Move to the latest Heroku stack (heroku-24)
 - Move production on Heroku to the latest Python (3.13.1)

We still need a `requirements.txt` file for Heroku, but it's now effectively a lock file generated using `hatch`.

### Feedback sought? <!-- What should reviewers focus on in particular? -->

General review

### Priority? <!-- How soon would you like this PR reviewed, does it block other work? -->

medium

### Tests added? <!-- Make sure your PR includes automated tests for your changes. -->

yes, CI now has a dedicated reproducing the full Heroku way to build

### How to test? <!-- Explain how reviewers should test this PR. -->

This version is currently deployed in preview mode on the heroku-24 stack at:

https://readalong-studio-heroku-24-488739ed89f8.herokuapp.com/api/v1/docs

(blocked as "Suspicious" on NRC networks, works elsewhere)

### Confidence? <!-- How confident are you that these changes are ready to merge? -->

pretty high

### Version change? <!-- Do you think this PR should trigger a Major (Breaking Change)/Minor (New Feature)/patch (refactor/bug fix) version change? -->

no

<!-- Add any other relevant information here -->

roedoejet commented on MANIFEST.in:
```diff
@@ -1,8 +0,0 @@
-recursive-include readalongs/templates *
-recursive-include readalongs/static *
```
do we not need to include readalongs/static? I agree the templates and epub folders aren't here anymore.

roedoejet replied:
I don't see the static folder listed in the pyproject file either, but maybe it just get's included automatically or something? has this been tested?

dhdaines replied:
Yes, Hatchling doesn't use `MANIFEST.in` from what I can tell: https://hatch.pypa.io/latest/why/#build-backend

So we need to double check to make sure the right things are being included in the source/wheel distributions.

joanise replied:
Good point, and thanks for the reminder @dhdaines! When Sam and I converted EveryVoice to `pyproject.toml`, we had to do some work to get the wheel and tar ball to contain the right files, and I didn't even test that yet with this PR. Better do that!

I've never understood what MANIFEST.in actually does, I have to admit... 🤷 I suppose I ought to!

joanise replied:
Checks done: the wheel and tar ball get generated correctly.

roedoejet commented on readme-heroku.md:
```diff
@@ -1,16 +1,26 @@
-   Heroku detects Python by default, but `runtime.txt` lets us specify/bump the version as needed;
- - `requirements.txt`: tells Heroku what our production dependencies are.
+Updating dependencies:
+ - Our dependencies are declared in `pyproject.toml`, edit this file to modify them.
```
```suggestion
 - Our dependencies are declared in `pyproject.toml`, edit `pyproject.toml` to modify them.
```

roedoejet replied:
"this file" is ambiguous

roedoejet reviewed (commented):
I've got a question about the MANIFEST.in file and whether it's necessary anymore. Also, while looking at the docs in the demo API that is spun up I noticed that the help message for the `/langs` endpoint is incorrect. It gives this as an example:
```

[
    {"code": "alq", names: { "alq": "Anishinaabemowin", "_": "Algonquin" }},
    {"code": "atj", names: { "atj": "Nehiromowin", "_": "Atikamekw" }},
    {"code": "fra", names: { "fra": "Français", "_": "French" }},
    ...
]
```
but Français, Anishinaabemowin, and Nehiromowin don't actually show up, and I can't see any other examples of this. Also - by this logic, why is `_` not `eng`?

joanise commented:
> the help message for the `/langs` endpoint

I fixed the langs help message.

`_` was meant to be the key for the default display name, which is usually but not always English, and the other keys for names localized in specific languages, but in practice we have not done any of this localization work, so it's scaffolding for a future feature we have not implemented. I've adjusted the help to make that clear.

![image](https://github.com/user-attachments/assets/6b794afb-50ec-4eb6-8ed5-2e16d5b8d7df)
